torch/csrc/jit/backends/backend.cpp

```diff
@@ -0,0 +1,58 @@
+#include <torch/csrc/jit/backends/backend.h>
+#include <torch/custom_class_detail.h>
+#include <torch/include/ATen/core/builtin_function.h>
+
+namespace torch {
+namespace jit {
+
+RegisterBackend::RegisterBackend(
+    std::string name,
+    BackendInterfaceCreator creator,
+    Preprocessor preprocess) {
+  // Make sure that the backend name is not qualified.
+  TORCH_CHECK(name.find('.') == std::string::npos);
+
+  // The preprocessor function will be registered under the name
+  // "torch.__backends__.__preprocess__.<backend_name>" and similarly, the
+  // creator function will be registered under the name
+  // "torch.__backends__.__creator__.<backend_name>".
+  const c10::QualifiedName backends_prefix("torch.__backends__");
+  const c10::QualifiedName creator_prefix(backends_prefix, "__creator__");
+  const c10::QualifiedName preprocess_prefix(backends_prefix, "__preprocess__");
+
+  // Arguments for the schemas of the creator and preprocessor functions.
+  c10::Argument mod("mod", c10::AnyType::get());
+  c10::Argument capsule("capule", c10::CapsuleType::get());
+  c10::Argument method_compile_spec(
+      "method_compile_spec",
+      c10::DictType::create(c10::StringType::get(), c10::AnyType::get()));
+
+  // Register the creator method.
+  c10::QualifiedName creator_qual_name(creator_prefix, name);
+  c10::FunctionSchema creator_schema(
+      name,
+      /*overload_name=*/"",
+      /*arguments=*/{mod},
+      /*returns=*/{capsule});
+  registerCustomClassMethod(std::make_unique<jit::BuiltinOpFunction>(
```

Comment:
`registerCustomClassMethod` doesn't check for name collisions. We should probably do that.